src/module/actor/SR5Actor.ts

```diff
@@ -199,178 +218,212 @@ export class SR5Actor extends Actor {
-            base: matrix.attack.base,
-            mod: matrix.attack.mod,
-            hidden: true,
-        };
-        limits.sleaze = {
-            value: matrix.sleaze.value,
-            base: matrix.sleaze.base,
-            mod: matrix.sleaze.mod,
-            hidden: true,
-        };
-
-        attributes.firewall = {
-            value: matrix.firewall.value,
-            base: matrix.firewall.base,
-            mod: matrix.firewall.mod,
-            hidden: true,
-        };
-        attributes.data_processing = {
-            value: matrix.data_processing.value,
-            base: matrix.data_processing.base,
-            mod: matrix.data_processing.mod,
-            hidden: true,
-        };
-        attributes.attack = {
-            value: matrix.attack.value,
-            base: matrix.attack.base,
-            mod: matrix.attack.mod,
-            hidden: true,
-        };
-        attributes.sleaze = {
-            value: matrix.sleaze.value,
-            base: matrix.sleaze.base,
-            mod: matrix.sleaze.mod,
-            hidden: true,
-        };
-
-        // SET ESSENCE
-        actorData.data.attributes.essence.value = +(totalEssence + modifiers['essence']).toFixed(3);
+    /**
+     * Prepare the actor data limits
+     * @param data
+     */
+    static prepareLimits(data: SR5ActorData) {
+        const { limits, attributes, modifiers } = data;
 
         // SETUP LIMITS
         limits.physical.value =
-            Math.ceil((2 * attributes.strength.value + attributes.body.value + attributes.reaction.value) / 3) + modifiers['physical_limit'];
-        limits.mental.value = Math.ceil((2 * attributes.logic.value + attributes.intuition.value + attributes.willpower.value) / 3) + modifiers['mental_limit'];
+            Math.ceil((2 * attributes.strength.value + attributes.body.value + attributes.reaction.value) / 3) + Number(modifiers['physical_limit']);
+        limits.mental.value =
+            Math.ceil((2 * attributes.logic.value + attributes.intuition.value + attributes.willpower.value) / 3) + Number(modifiers['mental_limit']);
         limits.social.value =
-            Math.ceil((2 * attributes.charisma.value + attributes.willpower.value + attributes.essence.value) / 3) + modifiers['social_limit'];
+            Math.ceil((2 * attributes.charisma.value + attributes.willpower.value + attributes.essence.value) / 3) + Number(modifiers['social_limit']);
 
-        // MOVEMENT
-        const movement = data.movement;
-        movement.walk.value = attributes.agility.value * (2 + modifiers['walk']);
-        movement.run.value = attributes.agility.value * (4 + modifiers['run']);
+        // limit labels
+        for (let [l, limit] of Object.entries(limits)) {
+            limit.label = CONFIG.SR5.limits[l];
+        }
+    }
+
+    /**
+     * Prepare actor data condition monitors (aka Tracks)
+     * @param data
+     */
+    static prepareConditionMonitors(data: SR5ActorData) {
```

Comment:
I don't like how this is only effecting and being called from an instance but is a static method. You're operating on instance data from a static context, which defeats the purpose of an static method.